### ISSUE TITLE
use color stack  is order dependant

### ISSUE THREAD

**Issue by u-fischer** [T-F-S/tcolorbox#157]
The `use color stack` key seems to work only if it is used *before* the breakable key. Imho this should be documented:

~~~~
\documentclass{article}
\usepackage{color,lipsum}
\usepackage{tcolorbox}
\tcbuselibrary{breakable}
\begin{document}
   \begin{tcolorbox}[breakable,use color stack] %doesn't work
     \color{red}
    \lipsum
    \end{tcolorbox}
 
\newpage
    \begin{tcolorbox}[use color stack,breakable] %works
    \color{red}
    \lipsum
    \end{tcolorbox}
\end{document}  
~~~~

**Comment by muzimuzhi** [comment]
Option `breakable` will call [`\tcb@init@breakability`](https://github.com/T-F-S/tcolorbox/blob/d853aebcc7de6787273fd418835d44faa77da5e9/tex/latex/tcolorbox/tcbbreakable.code.tex#L915-L926), which will define two internal hooks depending on the value of `\iftcb@usecolorstack`. But that would be too early when `use color stack` is not yet parsed.

The following example provides a short-term workaround. It's short-term because to make the change minimal I just moved the `\if..` condition into hook macros, which will slightly affect the efficiency.

```tex
\documentclass{article}
\usepackage{lipsum}
\usepackage{tcolorbox}
\tcbuselibrary{breakable}

\makeatletter
\def\tcb@init@breakability{%
  \let\tcb@savebox=\tcb@vbox%
  \let\endtcb@savebox=\endtcb@vbox%
  \let\tcb@drawcolorbox=\tcb@drawcolorbox@breakable%
  %%% change begin
  % delay \iftcb@usecolorstack
  \def\tcb@upperdraw@begin@hook{%
    \iftcb@usecolorstack
      \pdfcolSetCurrent{tcb@breakable}%
    \else
      \color{tcbcolupper}%
    \fi}%
  \def\tcb@lowerdraw@begin@hook{%
    \iftcb@usecolorstack
      \pdfcolSetCurrent{tcb@breakable}%
    \else
      \color{tcbcollower}%
    \fi}%
  %%% change end
}
\makeatother

\begin{document}
   \begin{tcolorbox}[breakable,use color stack] %doesn't work
     \color{red}
    \lipsum
    \end{tcolorbox}
 
\newpage
    \begin{tcolorbox}[use color stack,breakable] %works
    \color{red}
    \lipsum
    \end{tcolorbox}
\end{document}
```

**Comment by T-F-S** [comment]
I consider this to be a bug, since `use color stack` is not intended to be order dependant. A preview of the fix is uploaded to GitHub (not identical to @muzimuzhi but with the same idea to postpone the color stack decision)

**Comment by T-F-S** [comment]
Fixed with version `5.0.0 (2021-12-16)`.
